_pytest/monkeypatch.py

```diff
@@ -32,62 +31,71 @@ def pytest_funcarg__monkeypatch(request):
     return mpatch
 
 
+def resolve(name):
+    # simplified from zope.dottedname
+    parts = name.split('.')
 
-def derive_importpath(import_path, raising):
-    import pytest
-    if not isinstance(import_path, _basestring) or "." not in import_path:
-        raise TypeError("must be absolute import path string, not %r" %
-                        (import_path,))
-    rest = []
-    target = import_path
-    target_parts = set(target.split("."))
-    while target:
+    used = parts.pop(0)
+    found = __import__(used)
+    for part in parts:
+        used += '.' + part
         try:
-            obj = __import__(target, None, None, "__doc__")
+            found = getattr(found, part)
+        except AttributeError:
+            pass
+        else:
+            continue
+        # we use explicit un-nesting of the handling block in order
+        # to avoid nested exceptions on python 3
+        try:
+            __import__(used)
         except ImportError as ex:
-            if hasattr(ex, 'name'):
-                # Python >= 3.3
-                failed_name = ex.name
+            # str is used for py2 vs py3
+            expected = str(ex).split()[-1]
```

Comment:
is there a alternative to the `_totext` function? 
i recall python exceptions should be native strings